src/main/scala/cromwell/binding/RuntimeAttributes.scala

```diff
@@ -2,126 +2,217 @@ package cromwell.binding
 
 import com.google.api.services.genomics.model.Disk
 import cromwell.binding.AstTools.{AstNodeName, EnhancedAstNode}
-import cromwell.binding.RuntimeAttributes.{Defaults, LocalDisk}
 import cromwell.parser.RuntimeKey._
 import cromwell.parser.WdlParser.{Ast, AstList}
 import cromwell.parser.{BackendType, MemorySize, RuntimeKey}
 import org.slf4j.LoggerFactory
-
 import scala.collection.JavaConverters._
 import scala.language.postfixOps
+import scalaz._
+import Scalaz._
+import cromwell.binding.RuntimeAttributes._
+
+case class RuntimeAttributes(docker: Option[RuntimeDocker],
+              defaultZones: RuntimeZones,
+              failOnStderr: RuntimeFailOnStderr,
+              cpu: RuntimeCpu,
+              preemptible: RuntimePreemptible,
+              defaultDisks: RuntimeDisks,
+              memoryGB: RuntimeMemory)
 
 object RuntimeAttributes {
-  /** Convenience class to convert to Google `Disk` class. */
-  private case class LocalDisk(name: String, sizeGb: Long, diskType: String) {
-    def toDisk: Disk = new Disk().setSizeGb(sizeGb).setType(diskType).setName(name)
+  private val log = LoggerFactory.getLogger("RuntimeAttributes")
+
+  def apply(ast: Ast, backendType: BackendType): RuntimeAttributes = {
+    val attributeMap = ast.toAttributes
+
+    /**
+     *  Warn if keys are found that are unsupported on this backend.  This is not necessarily an error, at this point
+     *  the keys are known to be supported on at least one backend other than this.
+     */
+    attributeMap.unsupportedKeys(backendType) foreach log.warn
+
+    val attributeMapNel = validateAttributeMap(attributeMap, backendType)
+    val runtimeAttributeNel = validateRuntimeAttributes(attributeMap)
+    val validatedRuntimeAttributes = (attributeMapNel |@| runtimeAttributeNel) { (m, r) => r }
+
+    validatedRuntimeAttributes match {
+      case Success(r) => r
+      case Failure(f) =>
+        val errorMessages = f.list.mkString(", ")
+        throw new IllegalArgumentException(s"RuntimeAttribute is not valid: Errors: $errorMessages")
+    }
   }
 
-  private val log = LoggerFactory.getLogger("RuntimeAttributes")
+  sealed trait RuntimeAttribute
+  final case class RuntimeDocker(image: String) extends RuntimeAttribute
+  final case class RuntimeFailOnStderr(failOnStderr: Boolean) extends RuntimeAttribute
+  final case class RuntimeCpu(count: Long) extends RuntimeAttribute
+  final case class RuntimePreemptible(isPreemptible: Boolean) extends RuntimeAttribute
+  final case class RuntimeZones(zones: Seq[String]) extends RuntimeAttribute
+  final case class RuntimeDisks(disks: Seq[Disk]) extends RuntimeAttribute
+  final case class RuntimeMemory(gb: Double) extends RuntimeAttribute
```

Comment:
I'm wondering if these case classes are worth the trouble, they add a lot of repetitive dereferences and I'm not sure they provide much benefit other than the one spot constructing the `RuntimeAttributes`.